### DISCUSSION TITLE
local image can't be found, though it exists

### DISCUSSION THREAD

**Discussion by AISTALK** [slimtoolkit/slim#609] [Q&A]
Hello everyone!

I have a problem, I have no idea even what to try to solve it, for some reason docker-slim can't detect local image,  any help is welcome!

aitalent@AITalent-M1402IA:~/snap/code/streamlit_app$ docker images
REPOSITORY                  TAG       IMAGE ID       CREATED        SIZE
streamlit_room_detect_app   latest    17681b1d12ac   18 hours ago   6.12GB
dslim/slim                  latest    d3b74eb20a91   12 days ago    54.9MB

aitalent@AITalent-M1402IA:~/snap/code/streamlit_app$ docker-slim build --target streamlit_room_detect_app --tag streamlit_room_detect_app_slim
cmd=build info=param.http.probe message='using default probe' 
cmd=build state=started
cmd=build info=params target.type='image' target.image='streamlit_room_detect_app' continue.mode='probe' rt.as.user='true' keep.perms='true' tags='streamlit_room_detect_app_slim' image-build-engine='internal' 
cmd=build info=target.image.error status='image.not.found' image='streamlit_room_detect_app' message='make sure the target image already exists locally (use --pull flag to auto-download it from registry)' 
cmd=build state=exited code=16777218
cmd=build info=report file='slim.report.json' 
app='slim' message='GitHub Discussions' info='https://github.com/slimtoolkit/slim/discussions'
app='slim' message='Join the CNCF Slack channel to ask questions or to share your feedback' info='https://cloud-native.slack.com/archives/C059QP1RH1S'
app='slim' message='Join the Discord server to ask questions or to share your feedback' info='https://discord.gg/9tDyxYS'
app='slim' message='Join the Gitter channel to ask questions or to share your feedback' info='https://gitter.im/docker-slim/community'

**Comment by AISTALK** [comment]
slim.report.json btw

```
{
  "version": "1.1",
  "engine": "linux|Transformer|1.40.6|df98aa8c0e3ed3b15a25925cc07c1b54a5a6627a|2023-11-02_06:03:34PM",
  "containerized": false,
  "host_distro": {
    "name": "Ubuntu",
    "version": "22.04",
    "display_name": "Ubuntu 22.04.3 LTS"
  },
  "type": "build",
  "state": "error",
  "target_reference": "streamlit_room_detect_app",
  "system": {
    "type": "",
    "release": "",
    "distro": {
      "name": "",
      "version": "",
      "display_name": ""
    }
  },
  "source_image": {
    "identity": {
      "id": ""
    },
    "size": 0,
    "size_human": "",
    "create_time": "",
    "docker_version": "",
    "architecture": "",
    "container_entry": {
      "exe_path": ""
    }
  },
  "minified_image_size": 0,
  "minified_image_size_human": "",
  "minified_image": "",
  "minified_image_has_data": false,
  "minified_by": 0,
  "artifact_location": "",
  "container_report_name": "",
  "seccomp_profile_name": "",
  "apparmor_profile_name": "",
  "image_stack": null,
  "image_created": false,
  "image_build_engine": ""
}
```

**Comment by xies4** [comment]
Same issue here.

**Reply by kcq** [reply]
The new release, 1.40.7, has a number of enhancements to troubleshoot this and similar conditions. Give it a try! For example, there's a new command to list container images: `slim images`, which will show you what container images slim can see. The new version will also let you know if it encounters access auth related failures.

Quick summary of the problems:

1. Docker unix socket permission problem where the user account is not configured to be in the docker group.

2. Docker engine and Docker Desktop are installed on the same machine and 'slim' picks up one of them and you are looking at the other.

Quick summary of the solutions:

1. Double check the user permissions and I'd the user can access the Docker socket.

2. Double check the unix socket for the Docker setup and tell slim about it if it's not the default docker engine socket or an alias / symlink for it. Note that you can tell Docker Desktop to configure the global / default Docker socket. If you don't do that your Docker Desktop will have a user specific Docker socket.
